test/test_ious.py

```diff
@@ -0,0 +1,147 @@
+from typing import List, Callable
+
+import pytest
+import torch
+import torch.fx
+from torch import Tensor
+from torchvision import ops
+
+
+class IouTestBase:
+    @staticmethod
+    def _run_test(target_fn: Callable, test_input: List, dtypes: List[torch.dtype], tolerance: float, expected: List):
+        def assert_close(box: Tensor, expected: Tensor, tolerance):
+            out = target_fn(box, box)
+            torch.testing.assert_close(out, expected, rtol=0.0, check_dtype=False, atol=tolerance)
+
+        for dtype in dtypes:
+            actual_box = torch.tensor(test_input, dtype=dtype)
+            expected_box = torch.tensor(expected)
+            assert_close(actual_box, expected_box, tolerance)
+
+    @staticmethod
+    def _run_jit_test(target_fn: Callable, test_input: List):
+        box_tensor = torch.tensor(test_input, dtype=torch.float)
+        expected = target_fn(box_tensor, box_tensor)
+        scripted_fn = torch.jit.script(target_fn)
+        scripted_out = scripted_fn(box_tensor, box_tensor)
+        torch.testing.assert_close(scripted_out, expected, rtol=0.0, atol=1e-3)
+
+
+def _generate_int_input():
+    return [[0, 0, 100, 100], [0, 0, 50, 50], [200, 200, 300, 300]]
+
+
+def _generate_float_input():
+    return [
+        [285.3538, 185.5758, 1193.5110, 851.4551],
+        [285.1472, 188.7374, 1192.4984, 851.0669],
+        [279.2440, 197.9812, 1189.4746, 849.2019],
+    ]
+
+
+class TestBoxIou(IouTestBase):
+    def _generate_int_expected():
```

Comment:
not pretty happy with this. But there are 3 ways to work around as far as I know
1. Either create the function outside.
2. Global variable
3. Fixture.

Reply:
> 3. Fixture.

That is probably not a good option since we want to iterate over multiple cases, but fixtures in general only support one value. Of course we could also loop over the inputs in the test, but that is less expressive as there will be no information what input failed.

> 1. Either create the function outside.
> 2. Global variable

Given that these functions don't take any inputs, I think we can use a variable instead. Plus, can't we define that as class attribute like

```py
class TestFoo:
    FOOS = ["foo", "bar", "baz"]

    @pytest.mark.parametrize("foo", FOOS)
    def test_foo(self, foo):
        pass
```

?

Reply:
Adapted to these per class as the expected calculation differs for each type of IoU